src/base/SolutionArray.cpp

```diff
@@ -36,30 +36,229 @@ const std::map<std::string, std::string> aliasMap = {
 namespace Cantera
 {
 
-SolutionArray::SolutionArray(
-    const shared_ptr<Solution>& sol,
-    size_t size,
-    const AnyMap& meta)
+SolutionArray::SolutionArray(const shared_ptr<Solution>& sol,
+                             size_t size, const AnyMap& meta)
     : m_sol(sol)
     , m_size(size)
+    , m_dataSize(size)
     , m_meta(meta)
 {
     if (!m_sol) {
         throw CanteraError("SolutionArray::SolutionArray",
             "Unable to create SolutionArray from invalid Solution object.");
     }
     m_stride = m_sol->thermo()->stateSize();
-    m_data.resize(m_size * m_stride, 0.);
+    m_data.reset(new vector<double>(m_dataSize * m_stride, 0.));
+    m_extra.reset(new map<string, AnyValue>());
+    m_order.reset(new map<int, string>());
```

Comment:
In line with some style preferences introduced in #1452:
```suggestion
    m_data = make_shared<vector<double>>(m_dataSize * m_stride, 0.);
    m_extra = make_shared<map<string, AnyValue>>();
    m_order = make_shared<map<int, string>>();
```